src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCache.cs

```diff
@@ -21,7 +21,11 @@ internal abstract class AssetCache : ScriptableObject
         public static TAssetCache LoadAssetCache<TAssetCache>()
             where TAssetCache : AssetCache
         {
-            return Resources.Load<TAssetCache>(typeof(TAssetCache).Name);
+            using (StateSynchronizationPerformanceMonitor.Instance.IncrementEventDuration(typeof(TAssetCache).Name, "LoadAssetCache"))

```

Comment:
Will IncrementEventDuration and MeasureEventMemoryUsage be paired often? Looks like a dedicated method could be a perf win.

Reply:
I think at this point in time i'll hold off on a combined method.